driver/kubernetes/driver.go

```diff
@@ -101,22 +102,33 @@ func (d *Driver) wait(ctx context.Context) error {
 		err  error
 		depl *appsv1.Deployment
 	)
-	for try := 0; try < 100; try++ {
-		depl, err = d.deploymentClient.Get(ctx, d.deployment.Name, metav1.GetOptions{})
-		if err == nil {
-			if depl.Status.ReadyReplicas >= int32(d.minReplicas) {
-				return nil
-			}
-			err = errors.Errorf("expected %d replicas to be ready, got %d",
-				d.minReplicas, depl.Status.ReadyReplicas)
-		}
+	timeoutChan := time.After(d.provisioningTimeout)
+	ticker := time.NewTicker(100 * time.Millisecond)
+	tryCounter := 0
+	defer ticker.Stop()
+
+	for {
 		select {
+		case <-ticker.C:
+			if tryCounter < 100 {
+				tryCounter++
+			}
+			ticker.Stop()
+			ticker = time.NewTicker(time.Duration(100+tryCounter*20) * time.Millisecond)
```

Comment:
Im not sure if that logic is correct and if there is no better way of doing that. 
Or maybe I should keep it static as `100*time.Millisecond`